activesupport/lib/active_support/duration.rb

```diff
@@ -211,6 +210,7 @@ def calculate_total_seconds(parts)
     def initialize(value, parts) #:nodoc:
       @value, @parts = value, parts.to_h
       @parts.default = 0
+      parts.reject! { |k, v| v.zero? }
```

Comment:
This should be modifying `@parts` (the object's state), not `parts` (the initializer argument).

If `parts` is a hash then they're the same object and it doesn't matter, but sometimes it's an array-of-arrays:

https://github.com/rails/rails/blob/v5.1.4/activesupport/lib/active_support/duration.rb#L153

Which leads to this strange behaviour:

```ruby
0.minutes
=> 0 minutes
0.minutes + 0.minutes
=> 0 seconds
```